programs/uxd/src/mercurial_utils/calculate_possible_lp_token_precision_loss_collateral_value.rs

```diff
@@ -0,0 +1,25 @@
+use crate::{error::UxdError, utils::compute_precision_loss};
+use anchor_lang::{
+    prelude::{Account, Clock, SolanaSysvar},
+    Result,
+};
+
+use mercurial_vault::state::Vault;
+
+// Calculate how much collateral could be lost in possible LP token precision loss
+pub fn calculate_possible_lp_token_precision_loss_collateral_value(
+    mercurial_vault: &Account<Vault>,
+    mercurial_vault_lp_mint_supply: u64,
+) -> Result<u64> {
+    let current_time = u64::try_from(Clock::get()?.unix_timestamp)
+        .ok()
+        .ok_or(UxdError::MathError)?;
+
+    // Calculate the price of 1 native LP token
+    // Do not use mercurial_vault.get_amount_by_share because it does not handle precision loss
+    let total_amount = mercurial_vault
+        .get_unlocked_amount(current_time)
+        .ok_or(UxdError::MathError)?;
+
+    compute_precision_loss(total_amount, mercurial_vault_lp_mint_supply)
```

Comment:
I think you inverted the parameters here.

To avoid this mistake in the future, we can rename the parameters to:

- total_shares_amount -> total_shares_supply
- total_shares_value -> total_shares_value

the term "supply" would make it clearer for the first parameter, might also help with intelissense

Reply:
I think we could apply this to all other utility functions, "total_shares_supply" is much clearer than "total_shares_amount" but that's out of scope of this PR

Reply:
looks great otherwise!!

Reply:
renamed it on credix on another PR:
https://github.com/UXDProtocol/uxd-program/pull/223/files#diff-c012d070093436f787024d1947f0a8c45320f25fa52554b7d5aff38112adc76eL14

Reply:
should it even be renamed to `compute_shares_amount_for_value_ceiling` or sth? it's a math util function anyway and the formula itself isn't directly related to precision loss

Reply:
Yeah imo that's also a valid name for sure 👌

Reply:
I think the correct name is:

`compute_value_for_single_share_ceil`

Reply:
Since we're not calculating the "shares_amount" but we're computing the value of a single share rounded up